compiler/rustc_codegen_ssa/src/mir/mod.rs

```diff
@@ -199,6 +201,25 @@ pub fn codegen_mir<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
             }
         }
     }
+    if !all_consts_ok {
+        // There is no delay_span_bug here, we just have to rely on a hard error actually having
+        // been raised.
+        bx.abort();
+        // `abort` does not terminate the block, so we still need to generate
+        // an `unreachable` terminator after it.
+        bx.unreachable();
+        // We also have to delete all the other basic blocks again...
+        for bb in mir.basic_blocks().indices() {
+            if bb != mir::START_BLOCK {
+                unsafe {
+                    bx.delete_basic_block(fx.blocks[bb]);
```

Comment:
FWIW, `delete_basic_block` is unsafe but I was unable to find documentation for what the safety condition is. Cc @eddyb @nikomatsakis @bjorn3  ; who else could know about this?

Reply:
For cg_llvm it calls `LLVMDeleteBasicBlock` which calls `BasicBlock::eraseFromParent` which uses `vector::erase` to remove the basic block, thus invalidating all pointers to it. Other blocks may still reference it, so I don't think this is safe to do.

Reply:
What about checking if all constant could be evaluated first before creating any basic blocks?

Reply:
> Would it work if you just keep the basic blocks and never fill them?

I tried that first. Then I get assertion failures about missing terminators.
Maybe I should somehow add an `unreachable` terminator to each basic block... this is so silly.^^

Reply:
> Maybe I should somehow add an unreachable terminator to each basic block

Actually, that does not seem possible. There seems to be no API to switch the build to a different basic block.

Reply:
> I tried that first. Then I get assertion failures about missing terminators.

Then it probably tries to create an object file even when there were compilation errors before.

Reply:
Possible. FWIW, the assertion failures only showed on CI, not locally (so they probably require llvm-debug-assert or so).